core/client/fs/src/main/java/alluxio/client/metrics/ClientMasterSync.java

```diff
@@ -85,4 +94,32 @@ public synchronized void heartbeat() {
       SAMPLING_LOG.warn("Failed to send metrics to master: {}", e.toString());
     }
   }
+
+  /**
+   * Close the metrics master client.
+   */
+  public synchronized void close() {
+    if (mMasterClient != null) {
+      mMasterClient.close();
+    }
+  }
+
+  /**
+   * Loads configuration and initialize metrics master client.
+   */
+  private void loadConfAndInitClient() {
+    try {
+      InetSocketAddress masterAddr = mInquireClient.getPrimaryRpcAddress();
+      mContext.loadConf(masterAddr, true, true);
+    } catch (UnavailableException e) {
+      SAMPLING_LOG.error("Failed to get master address during initialization: {}", e.toString());
+    } catch (AlluxioStatusException ae) {
+      SAMPLING_LOG.error("Failed to load configuration from "
+          + "meta master during initialization: {}", ae.toString());
+    }
+    mMasterClient = new RetryHandlingMetricsMasterClient(MasterClientContext
```

Comment:
Should we still create `mMasterClient` when we fail above? Since heartbeat will be called periodically, it might be easier to just no-op this heartbeat and try again next time?

Reply:
Make sense, now retry load conf if fail and make heartbeat no-op if load conf failed